src/main/java/frc/trigon/robot/subsystems/ledstrip/LEDStrip.java

```diff
@@ -1,189 +1,97 @@
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import edu.wpi.first.wpilibj2.command.button.Trigger;
-import frc.trigon.robot.Robot;
 import frc.trigon.robot.commands.factories.GeneralCommands;
 
-import java.awt.*;
-import java.util.function.Function;
-
 public class LEDStrip extends SubsystemBase {
-    private static final CANdle CANDLE = LEDStripConstants.CANDLE;
-    private static final Trigger LOW_BATTERY_TRIGGER = new Trigger(() -> !DriverStation.isEnabled() && Robot.IS_REAL && RobotController.getBatteryVoltage() < LEDStripConstants.MINIMUM_BATTERY_VOLTAGE);
-    private static int LAST_CREATED_LED_STRIP_ANIMATION_SLOT = 0;
-    private static int LAST_STRIP_INDEX = 8;
-    private final int animationSlot;
-    private final int offset, numberOfLEDs;
+    private static final AddressableLED LED = LEDStripConstants.LED;
+    private final int indexOffset;
     private final boolean inverted;
+    private final int numberOfLEDs;
+    private double rainbowFirstPixelHue = 0;
+    private boolean areLEDsOnForBlinking = false;
+    private double lastBlinkTime = 0;
 
     static {
-        GeneralCommands.getDelayedCommand(1, () -> LOW_BATTERY_TRIGGER.whileTrue(LEDStripCommands.getAnimateSingleFadeCommand(Color.red, LEDStripConstants.LOW_BATTERY_FLASHING_SPEED, LEDStripConstants.LED_STRIPS))).schedule();
+        GeneralCommands.getDelayedCommand(
+                1,
+                () -> LEDStripConstants.LOW_BATTERY_TRIGGER.whileTrue(LEDStripCommands.getBlinkingCommand(Color.kRed, LEDStripConstants.SHOULD_LOW_BATTERY_LEDS_BLINK_FAST))
+        );
     }
 
-    /**
-     * Constructs a new LEDStrip.
-     *
-     * @param numberOfLEDs the number of LEDs in the strip
-     * @param inverted     is the strip inverted
-     */
-    public LEDStrip(int numberOfLEDs, boolean inverted) {
-        this.offset = LAST_STRIP_INDEX;
-        this.numberOfLEDs = numberOfLEDs;
+    public LEDStrip(boolean inverted, int numberOfLEDs, int indexOffset) {
+        this.indexOffset = indexOffset;
         this.inverted = inverted;
-
-        LAST_STRIP_INDEX = offset + numberOfLEDs;
-        LAST_CREATED_LED_STRIP_ANIMATION_SLOT++;
-        animationSlot = LAST_CREATED_LED_STRIP_ANIMATION_SLOT;
-    }
-
-    public static void setDefaultCommandForAllLEDS(Function<LEDStrip, Command> command) {
-        for (LEDStrip ledStrip : LEDStripConstants.LED_STRIPS)
-            ledStrip.setDefaultCommand(command.apply(ledStrip));
+        this.numberOfLEDs = numberOfLEDs;
     }
 
-    void staticColor(Color color) {
-        CANDLE.setLEDs(color.getRed(), color.getGreen(), color.getBlue(), 0, offset, numberOfLEDs);
+    public static void setDefaultCommandForAllLEDS(Command command) {
+        LEDStripConstants.RIGHT_CLIMBER_LEDS.setDefaultCommand(command);
+        LEDStripConstants.LEFT_CLIMBER_LEDS.setDefaultCommand(command);
```

Comment:
There might be more than two leds

Reply:
Then i'll add the other LEDs here

Reply:
Make this more robust and that it works in every situation where another led is created automatically